File: VB/WinSolution.Module.Win/CustomPrintingController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.XtraPrinting
Imports DevExpress.ExpressApp.Printing.Win

Namespace WinSolution.Module.Win
	Public Class CustomPrintingController
		Inherits PrintingController
		Protected Overrides Sub OnPrintingSettingsLoaded(ByVal args As PrintableComponentLinkEventArgs)
			MyBase.OnPrintingSettingsLoaded(args)
			CustomizePrintingSystem(args.PrintableComponentLink)
		End Sub
		Private Sub CustomizePrintingSystem(ByVal link As PrintableComponentLink)
			AddHandler link.CreateMarginalHeaderArea, AddressOf link_CreateMarginalHeaderArea
			AddHandler link.CreateMarginalFooterArea, AddressOf link_CreateMarginalFooterArea
		End Sub
		Private Sub link_CreateMarginalFooterArea(ByVal sender As Object, ByVal e As CreateAreaEventArgs)
			e.Graph.DrawPageInfo(PageInfo.DateTime, "{0:hhhh:mmmm:ssss}", Color.Green, New RectangleF(0, 0, 200, 20), BorderSide.None)
			e.Graph.DrawString("Custom text in the footer", Color.Red, New RectangleF(200, 0, 200, 20), BorderSide.None)
		End Sub
		Private Sub link_CreateMarginalHeaderArea(ByVal sender As Object, ByVal e As CreateAreaEventArgs)
			e.Graph.DrawPageInfo(PageInfo.DateTime, "{0:hhhh:mmmm:ssss}", Color.Green, New RectangleF(0, 0, 200, 20), BorderSide.None)
			e.Graph.DrawString("Custom text in the header", Color.Red, New RectangleF(200, 0, 200, 20), BorderSide.None)
		End Sub
	End Class
End Namespace

File: CS/E1229/WinSolution.Module.Win/CustomPrintingController.cs
using System;
using System.Drawing;
using DevExpress.XtraPrinting;
using DevExpress.ExpressApp.Printing.Win;

namespace WinSolution.Module.Win {
    public class CustomPrintingController : PrintingController {
        protected override void OnPrintingSettingsLoaded(PrintableComponentLinkEventArgs args) {
            base.OnPrintingSettingsLoaded(args);
            CustomizePrintingSystem(args.PrintableComponentLink);
        }
        private void CustomizePrintingSystem(PrintableComponentLink link) {
            link.CreateMarginalHeaderArea += link_CreateMarginalHeaderArea;
            link.CreateMarginalFooterArea += link_CreateMarginalFooterArea;
        }
        private void link_CreateMarginalFooterArea(object sender, CreateAreaEventArgs e) {
            e.Graph.DrawPageInfo(PageInfo.DateTime, "{0:hhhh:mmmm:ssss}", Color.Green, new RectangleF(0, 0, 200, 20), BorderSide.None);
            e.Graph.DrawString("Custom text in the footer", Color.Red, new RectangleF(200, 0, 200, 20), BorderSide.None);
        }
        private void link_CreateMarginalHeaderArea(object sender, CreateAreaEventArgs e) {
            e.Graph.DrawPageInfo(PageInfo.DateTime, "{0:hhhh:mmmm:ssss}", Color.Green, new RectangleF(0, 0, 200, 20), BorderSide.None);
            e.Graph.DrawString("Custom text in the header", Color.Red, new RectangleF(200, 0, 200, 20), BorderSide.None);
        }
    }
}